### DISCUSSION TITLE
olimex_esp32_poe - keine Ethernetverbindung mit OpenDTU_VeDirect, mit OpenDTU klappt es aber

### DISCUSSION THREAD

**Discussion by bagsbanni** [hoylabs/OpenDTU-OnBattery#98] [Q&A] [answered]
Moin zusammen,

ein sehr schönes Projekt. Leider habe ich ein kleines Problem mit dem olimex_esp32_poe. Flashe ich OpenDTU_VeDirect drauf (mit default_envs = olimex_esp32_poe), bekomme ich keine Ethernet-Verbindung (taucht das ESP nicht im DHCP-Server auf). Flashe ich hingegen OpenDTU (mit default_envs = olimex_esp32_poe), erscheint das ESP im DHCP-Server und erhält auch eine IP-Adresse zugewiesen. Bereits mehrmals reproduziert, mit OpenDTU klappt Ethernet, mit OpenDTU_VeDirect nicht.

Ich hatte schon testweise mit dem Tool winmerge die beiden Verzeichnisse OpenDTU und OpenDTU_VeDirect auf Unterschiede in den Dateien verglichen, bin aber noch nicht auf die Ursache gestossen.

Hat jemand eine Idee, woran es liegen könnte? Wäre für hilfreiche Hinweise dankbar ;O)

LG

**Comment by helgeerbe** [comment, marked as answer]
Hi @bagsbanni,

den olimex_esp32_poe kenne ich nicht. Hast du denn in der Config von olimex_esp32_poe die Victron Pins gesetzt (die hatte ich nicht angepasst, das ich so ein Bord nicht habe)?

```
[env:olimex_esp32_poe]
; https://www.olimex.com/Products/IoT/ESP32/ESP32-POE/open-source-hardware
board = esp32-poe
build_flags = ${env.build_flags}
    -DHOYMILES_PIN_MISO=15
    -DHOYMILES_PIN_MOSI=2
    -DHOYMILES_PIN_SCLK=14
    -DHOYMILES_PIN_IRQ=13
    -DHOYMILES_PIN_CE=16
    -DHOYMILES_PIN_CS=5
    -DOPENDTU_ETHERNET
````

Wenn du sie nicht gesetzt hast, wird defaultmäßig GPIO21 und GPIO22 verwendet. Wenn ich mir das Pinout des Boards im Internet anschaue, werden diese GPIOs nicht als Pins rausgeführt. Ich vermute stark, dass sie intern für den Ethernetport verwendet werden. Du könntest auf GPIO32 und GPIO33 ausweichen, die in dem Pinout, dass ich gefunden habe als "free to use" bezeichnet werden.

````
	-DVICTRON_PIN_TX=32
	-DVICTRON_PIN_RX=33
````

Ich habe auch den DeviceManager angepasst. Die Victron TX/RX PINS werden jetzt angezeigt. Wenn ich alle nötigen Stellen gefunden habe, müsste auch das Profil geladen werden können. Dann könntest du die Pins ohne Neuübersetzung austesten. (Von meiner Seite aber ungetesteter Code).

Das Json File müsste ungefähr so aussehen:

````
[
    {
        "name": "olimex_esp32_poe victron",
        "victron": {
            "rx": 33,
            "tx": 32
        }
    }
]
````

**Comment by bagsbanni** [comment]
Moin @helgeerbe,

vielen Dank für die schnelle Antwort. Ich hatte die Pins bereits aufs Blaue hinein (ich habe den olimex noch nicht mit dem Victron verbunden, da ich damit vorerst noch rumteste) auf    
-DVICTRON_PIN_TX=32
-DVICTRON_PIN_RX=33    
gesetzt, da ich sonst  wg. Kompilierungsfehler gar keinen Upload hätte durchführen können.

Ich werde dann heute Abend mal neu Clonen und erneut testen. Ich gebe Rückmeldung, ob Ethernet damit dann klappt.

**Comment by bagsbanni** [comment]
Moin @helgeerbe,

habe es jetzt schon mal in meiner Mittagspause getestet, damit läuft die Ethernetverbindung nun auch!

Vielen Dank.
